matlab/FeatherReader.cpp

```diff
@@ -0,0 +1,301 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <arrow/io/file.h>
+#include <arrow/ipc/feather.h>
+#include <arrow/status.h>
+#include <arrow/table.h>
+#include <arrow/type.h>
+
+#include <mex.h>
+
+#include "FeatherReader.h"
+#include "FeatherType.h"
+
+namespace matlab {
+namespace arrow {
+namespace ipc {
+namespace feather {
```

Comment:
These extra 3 namespaces serve to complicate things. 

* Does the `matlab` namespace help (are there any `matlab::` symbols?)? If not, then perhaps have a single namespace `mlarrow` or similar